browsing-context.md

```diff
@@ -1,28 +1,295 @@
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+## Table of contents
+
+- [Example](#example)
+- [Restrictions](#restrictions)
+  - [Privacy-based restrictions](#privacy-based-restrictions)
+    - [Storage access blocking](#storage-access-blocking)
+    - [Communications channels that are blocked](#communications-channels-that-are-blocked)
+    - [Communications channels that match navigation](#communications-channels-that-match-navigation)
+  - [Restrictions on the basis of being non-user-visible](#restrictions-on-the-basis-of-being-non-user-visible)
+  - [Restrictions on loaded content](#restrictions-on-loaded-content)
+- [JavaScript API](#javascript-api)
+  - [Current proposal](#current-proposal)
+  - [Adjacent APIs](#adjacent-apis)
+- [Session history](#session-history)
+- [Navigation](#navigation)
+- [Rendering-related behavior](#rendering-related-behavior)
+- [CSP integration](#csp-integration)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Example
+
+Consider `https://a.example/`, which contains the following HTML:
+
+```html
+<link rel="prerender2" href="https://b.example/">
+
+<a href="https://b.example/">Click me!</a>
+```
+
+_The `"prerender2"` rel here is illustrative only. See the [triggers document](./triggers.md) for more serious discussion of potential APIs for triggering prerendering._
+
+Upon loading `https://a.example/`, the browser notices the request to prerender `https://b.example/`. It does so by creating a prerendering browsing context, which it navigates to `https://b.example/`. This navigation takes place using [special fetch modes](./fetch.md), which ensure that `https://b.example/` has [opted in](./opt-in.md) to being prerendered, and ensures that the request for `https://b.example/` and any of its subresources is performed without any credentials that might identify the user.
+
+Within this prerendering browsing context, assuming the opt-in check passes, loading of `https://b.example/` proceeds mostly as normal. This includes any expensive web-developer-provided JavaScript necessary to initialize the web app found there. It could even include server- or client-side redirects to other pages, perhaps even other domains.
+
+However, if `https://b.example/` is one of those sites that requests notification permissions on first load, such a permission prompt will be denied, as if the user had declined. (TODO or should it defer prompting the user until activation, so that the relevant promise doesn't settle until then?) Similarly, if `https://b.example/` performs an `alert()` call, the call will instantly return, without the user seeing anything. Another key difference is that `https://b.example/` will not have any storage access, including to cookies. Thus, the content it initially renders will be a logged-out view of the web app, or perhaps a specially-tailored "prerendering" view which leaves things like logged-in state indeterminate.
```

Comment:
"will not have any (unpartitioned) storage access"

to be consistent with the statement below?

Reply:
I think it's more accurate in this section to say that you won't have any storage access, since that's the case. Below we're discussing the subtleties about unpartitioned vs. partitioned, but here we're discussing the whole landscape.